File: Java/14.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        int min = Integer.MAX_VALUE,count = 0,flag = 0;
        char chr;
        for(int i = 0; i < strs.length; i++){
            if(min > strs[i].length()) min = strs[i].length();
        }
        for(int i = 0; i < min; i++){
            flag = 0;
            chr = strs[0].charAt(i);
            for(int j = 1; j < strs.length; j++){
                if(strs[j].charAt(i) != chr){
                    flag = 1;
                    break;
                }
            }
            if(flag == 0) count++;
            else break;
        }
        return strs[0].substring(0,count);
    }
}

File: Python/14.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        it = min(len(i) for i in strs)
        l = len(strs)
        length = 0
        for i in range(it):
            flag = 1
            for j in range(1,l):
                if strs[j][:i+1] != strs[j-1][:i+1]:
                    flag = 0
                    break
            length += flag
            if flag == 0:
                break
        return strs[0][:length]